internal/querynodev2/delegator/delegator_data.go

```diff
@@ -1011,12 +1011,12 @@ func (sd *shardDelegator) buildBM25IDF(req *internalpb.SearchRequest) (float64,
 	proto.Unmarshal(req.GetPlaceholderGroup(), pb)
 
 	if len(pb.Placeholders) != 1 || len(pb.Placeholders[0].Values) == 0 {
-		return 0, merr.WrapErrParameterInvalidMsg("please provide varchar for BM25 Function based search")
+		return 0, merr.WrapErrParameterInvalidMsg("please provide varchar/text for BM25 Function based search")
 	}
 
 	holder := pb.Placeholders[0]
 	if holder.Type != commonpb.PlaceholderType_VarChar {
-		return 0, merr.WrapErrParameterInvalidMsg(fmt.Sprintf("please provide varchar for BM25 Function based search, got %s", holder.Type.String()))
+		return 0, merr.WrapErrParameterInvalidMsg(fmt.Sprintf("please provide varchar/text for BM25 Function based search, got %s", holder.Type.String()))
```

Comment:
NIT: text is not involved with placeholder here

Reply:
will fix in next PR, placeholder is always varchar type